arm_isa.cpp

```diff
@@ -1377,7 +1377,7 @@ inline void LDRH(arm_isa* ref, int rd, int rn,
     printf("Unpredictable LDRH instruction result (Address is not Halfword Aligned)\n");
     return;
   }
-  value = MEM.read(ref->ls_address.entire);
+  value = MEM.read_half(ref->ls_address.entire);
   value &= 0xFFFF; /* Zero extends halfword value 
 		      BUG: Model must be little endian in order to the code work  */
```

Comment:
I'm curious about the bug you found. Does it really crash the application? I'm saying this because this code was weird and inappropriate, but it should work for little endian hosts. By all means, your commit leads to better code and should land, but I was just curious to understand why it crashed anything at all.